crates/net/eth-wire/src/types/transactions.rs

```diff
@@ -94,6 +107,81 @@ impl BlobTransaction {
     }
 }
 
+#[cfg(any(test, feature = "arbitrary"))]
+impl<'a> arbitrary::Arbitrary<'a> for BlobTransaction {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let mut arr = [0u8; BYTES_PER_BLOB];
+        let blobs: Vec<Blob> = (0..u.int_in_range(1..=16)?)
+            .map(|_| {
+                arr = arbitrary::Arbitrary::arbitrary(u).unwrap();
+
+                // Ensure that the blob is canonical by ensuring that
+                // each field element contained in the blob is < BLS_MODULUS
+                for i in 0..(FIELD_ELEMENTS_PER_BLOB as usize) {
+                    arr[i * BYTES_PER_FIELD_ELEMENT] = 0;
+                }
+                Blob::from(arr)
+            })
+            .collect();
+
+        Ok(generate_blob_transaction(blobs, TransactionSigned::arbitrary(u)?))
+    }
+}
+
+#[cfg(any(test, feature = "arbitrary"))]
+impl proptest::arbitrary::Arbitrary for BlobTransaction {
+    type Parameters = ParamsFor<String>;
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (
+            proptest_vec(proptest_vec(proptest_any::<u8>(), BYTES_PER_BLOB), 1..=5),
```

Comment:
using `proptest_any::<[u8;  BYTES_PER_BLOB]>()`   instead of `proptest_vec(proptest_any::<u8>(), BYTES_PER_BLOB)` causes proptest to hit:

```
thread 'types::transactions::BlobTransactionTests::proptest' has overflowed its stack
```

Reply:
interesting, `BYTES_PER_BLOB` is pretty large, so i wonder if that's partially why